components/script/script_runtime.rs

```diff
@@ -798,27 +799,46 @@ impl StreamConsumer {
         }
     }
 
-    fn stream_error(&self, errorCode: usize){
+    fn stream_error(&self, error_code: usize){
         unsafe {
-            StreamConsumerStreamError(self.0, errorCode);
+            StreamConsumerStreamError(self.0, error_code);
         }
     }
 
     fn note_response_urls(
         &self,
-        maybeUrl: *const c_char,
-        maybeSourceMapUrl: *const c_char
+        maybe_url: Option<String>,
+        maybe_source_map_url: Option<String>
     ) {
         unsafe {
-            StreamConsumerNoteResponseURLs(self.0, maybeUrl, maybeSourceMapUrl);
+            let maybe_url_param: *const c_char = match maybe_url {
+                Some(maybe_url) => {
+                    let c_str_url = CString::new(maybe_url).unwrap();
+                    c_str_url.as_ptr()
```

Comment:
Unfortunately this isn't safe, since we store a pointer to a value that is destroyed when we leave this match scope. We need to do the following instead:
```rust
let maybe_url = maybe_url.map(|url| CString::new(url).unwrap());
let maybe_url_param = match maybe_url.as_ref() {
    Some(url) => url.as_ptr(),
    None => ptr::null(),
};
```

Reply:
Yeah. You're right. I'll make the changes.

Would it also work if I first convert Option<String> to Option<CString>, then just do the match?

```rust
let maybe_url = maybe_url.as_ref().map(|url| CString::new(url).unwrap());
let maybe_url_param = match maybe_url {
    Some(url) => url.as_ptr(),
    None => ptr::null(),
};
```

Reply:
I believe so!